instrumentation/README.md

```diff
@@ -11,7 +11,7 @@
 | [opentelemetry-instrumentation-boto3sqs](./opentelemetry-instrumentation-boto3sqs) | boto3 ~= 1.0 | No
 | [opentelemetry-instrumentation-botocore](./opentelemetry-instrumentation-botocore) | botocore ~= 1.0 | No
 | [opentelemetry-instrumentation-celery](./opentelemetry-instrumentation-celery) | celery >= 4.0, < 6.0 | No
-| [opentelemetry-instrumentation-confluent-kafka](./opentelemetry-instrumentation-confluent-kafka) | confluent-kafka >= 1.8.2, < 2.0.0 | No
+| [opentelemetry-instrumentation-confluent-kafka](./opentelemetry-instrumentation-confluent-kafka) | confluent-kafka >= 1.8.2, <= 2.1.1 | No
```

Comment:
Do you know why it was limited to 2.0.0 in the first place?

Reply:
It was limited to the latest major version back then

Reply:
But if there isn't a real reason to limit the version, maybe we should consider removing this,
because in the future we have to open PR just to update the limitation

Reply:
It's not a best practice to leave it on latest. Perhaps future versions won't be supported, tests won't pass and changes will be needed.

Reply:
It's possible that on newer versions of confluent-kafka the instrumentation will be broken, and without limitation for highest version possibly there will be a bug.

Reply:
To avoid the continuous overhead for having to change this value if they release a patch, I would change it to `< 2.2.0` We can trust that patch versions won't break the API interface

Reply:
@davidgonor1408 I am also interested on this, so please check my comment and let's see if we can get it merged soon :)

Reply:
Done